### PR TITLE
HB-6089: Update demo apps to support adaptive banners

### PR DESCRIPTION
**Pull request by kendallrogers**
Swift Demo:
Simulator Screen Shot - iPhone 14 Pro - 2023-10-13 at 14 12 55

SwiftUI:
Simulator Screen Shot - iPhone 14 Pro - 2023-10-13 at 14 12 36

### CODE REVIEW

File: swift/ChartboostMediationDemo/Ads/Banner/BannerAdViewController.swift [hunk 1]
```diff
@@ -20,22 +20,18 @@ import UIKit
 ///
 class BannerAdViewController: UIViewController {
 
-    /// The banner's container view. In the storyboard, it is constrained to be 320x50 for the sake of this
-    /// demo, which only uses the fixed `standard` size.
-    @IBOutlet private var bannerContainer: UIView!
```
Comment (kendallrogers) [inline swift/ChartboostMediationDemo/Ads/Banner/BannerAdViewController.swift:25]: I opted to remove the container and just add the banner to the view controller's view, which I feel better demonstrates adaptive banners.

File: swift/ChartboostMediationDemo/Ads/Fullscreen/FullscreenAdController.swift [hunk 2]
```diff
@@ -71,15 +71,15 @@ class FullscreenAdController: NSObject {
                 ad.delegate = self
                 self.fullscreenAd = ad
 
-                log(action: "load", placementName: placementName, error: result.error)
+                self.log(action: "load", placementName: self.placementName, error: result.error)
```
Comment (kendallrogers) [inline swift/ChartboostMediationDemo/Ads/Fullscreen/FullscreenAdController.swift:74]: `self` was missing here and causing the build to fail.

File: swiftui/ChartboostMediationDemo/Ads/Fullscreen/FullscreenAdController.swift [hunk 3]
```diff
@@ -58,17 +58,17 @@ class FullscreenAdController: NSObject, ObservableObject {
                 ad.delegate = self
                 self.fullscreenAd = ad
 
-                log(action: "Load", placementName: placementName, error: result.error)
+                self.log(action: "Load", placementName: self.placementName, error: result.error)
```
Comment (kendallrogers) [inline swiftui/ChartboostMediationDemo/Ads/Fullscreen/FullscreenAdController.swift:61]: `self` was missing here and causing the build to fail.

File: swiftui/ChartboostMediationDemo/Ads/Banner/BannerAdView.swift [hunk 4]
```diff
@@ -59,13 +59,11 @@ struct BannerAdView: View {
...
 
-                if controller.shouldShow, let bannerAd = controller.bannerAd {
+                if controller.shouldShow, let bannerAd = controller.bannerAd, let size = bannerAd.size?.size {
                     Spacer()
                     HStack {
-                        Spacer()
                         Banner(source: bannerAd)
-                            .frame(width: 320, height: 50)
-                        Spacer()
+                            .frame(width: size.width, height: size.height)
```
Comment (kendallrogers) [inline swiftui/ChartboostMediationDemo/Ads/Banner/BannerAdView.swift:66]: There is a `sizeThatFits` method on `UIViewRepresentable`, and using that allows us to give `Banner` the same intrinsic size as `ChartboostMediationBannerView`, however it's only available on iOS 16+, so manually setting the size of the `Banner` will have to do for now.

**Review by daniel-barros — APPROVED** [review]

**Comment by kendallrogers** [comment]
> LGTM Shouldn't we change the SDK dependency version in the podfile to point to 4.6 @kendallrogers ?

Yeah makes sense, since the new API won't work with any lower version. I just updated the Podfiles.

**Pull request merged**